app/models/downloaded_file.rb

```diff
@@ -1,47 +1,22 @@
 class DownloadedFile < ApplicationRecord
   class << self
-
-    def add_file(checksum_type, checksum, local_path)
-      return unless local_path.match?(/\.(rpm|drpm)$/)
-
-      file_size = File.size(local_path)
-      DownloadedFile.find_or_create_by({ checksum_type: checksum_type,
-                                         checksum: checksum,
-                                         local_path: local_path,
-                                         file_size: file_size })
-    end
-
-    def get_local_path_by_checksum(checksum_type, checksum)
-      DownloadedFile.find_by({ checksum_type: checksum_type, checksum: checksum })
+    def track_file(checksum_type:, checksum:, local_path:, size:)
+      find_or_initialize_by(local_path: local_path).tap do |record|
+        record.checksum_type = checksum_type
+        record.checksum = checksum
+        record.file_size = size
+
+        record.save if record.changed?
+      end.persisted?
```

Comment:
In fact, the return value is not used. I guess when I first thought about this function, since `model#save` doesn't throw an error, it could be useful to verify if the "track file" transaction was successful, but I don't verify it anywhere.

I don't know if RMT should really care about this, because unless the connection with the database fails, `model#save` should not fail; I used `#find_or_initialize_by` to avoid uniqueness validation errors.

I think it's safe to remove this `persisted?` by now. Any thoughts?

Reply:
I think it makes sense to remove it, because right now it's just overhead since it's not being used.

Reply:
Removed it :heavy_check_mark: